examples/src/bin/btc/main.rs

```diff
@@ -0,0 +1,160 @@
+    if let Some(data_dir) = args.get(1) {
+
+        let test_v = calculate_hash(&"139eeGkMGR6F9EuJQ3qYoXebfkBbNAsLtV:btc");
+
+        let path: PathBuf = [data_dir, "graphdb.bincode"].iter().collect();
+        let graph = if path.exists() {
+            let now = Instant::now();
+            let g = GraphDB::load_from_file(path.as_path()).expect("Failed to load graph");
+
+            println!(
+                "Loaded graph from path {} with {} vertices, {} edges, took {} seconds",
+                path.to_str().unwrap(),
+                g.len(),
+                g.edges_len(),
+                now.elapsed().as_secs()
+            );
+            g
+        } else {
+            let g = GraphDB::new(16);
+
+            let now = Instant::now();
+
+            let _ = CsvLoader::new(data_dir)
+                .with_filter(Regex::new(r".+(sent|received)").unwrap())
+                .load_into_graph(&g, |sent: Sent, g: &GraphDB| {
+                    let src = calculate_hash(&sent.addr);
+                    let dst = calculate_hash(&sent.txn);
+                    let time = sent.time.timestamp();
+
+                    if src == test_v || dst == test_v {
+                        println!("{} sent {} to {}", sent.addr, sent.amount_btc, sent.txn);
+                    }
+
+                    g.add_edge(
+                        src,
+                        dst,
+                        time.try_into().unwrap(),
+                        &vec![("amount".to_string(), Prop::U64(sent.amount_btc))],
+                    )
+                })
+                .expect("Failed to load graph");
+
+            println!(
+                "Loaded {} vertices, {} edges, took {} seconds",
+                g.len(),
+                g.edges_len(),
+                now.elapsed().as_secs()
+            );
+
+            g.save_to_file(path).expect("Failed to save graph");
+
+            g
+        };
+
+        assert!(graph.contains(test_v));
+        let deg_out = graph
+            .neighbours_window(0, i64::MAX, test_v, Direction::OUT)
+            .count();
+        let deg_in = graph
+            .neighbours_window(0, i64::MAX, test_v, Direction::IN)
+            .count();
+
+        println!(
+            "{} has {} out degree and {} in degree",
+            test_v, deg_out, deg_in
+        );
+    }
+}
+
+mod custom_date_format {
+    use chrono::{DateTime, TimeZone, Utc};
+    use serde::{self, Deserialize, Deserializer, Serializer};
+
+    const FORMAT: &'static str = "%Y-%m-%d %H:%M:%S";
+
+    // The signature of a serialize_with function must follow the pattern:
+    //
+    //    fn serialize<S>(&T, S) -> Result<S::Ok, S::Error>
+    //    where
+    //        S: Serializer
+    //
+    // although it may also be generic over the input types T.
+    pub fn serialize<S>(date: &DateTime<Utc>, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        let s = format!("{}", date.format(FORMAT));
```

Comment:
Not sure if we need a `date.unwrap()` here to make sure we fail before the format. Where do we get the initial date object from?

Reply:
taken care by library